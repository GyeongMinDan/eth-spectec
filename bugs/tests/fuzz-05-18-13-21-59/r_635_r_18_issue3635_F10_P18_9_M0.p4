# 1 "fuzz/fuzz1/illtyped/r_18_issue3635_F10_P18_9_M0.p4"
# 1 "<built-in>"
# 1 "<command-line>"
# 1 "fuzz/fuzz1/illtyped/r_18_issue3635_F10_P18_9_M0.p4"
# 12 "fuzz/fuzz1/illtyped/r_18_issue3635_F10_P18_9_M0.p4"
enum bit<4> e {
a = ((bool) ({ })),
b = 2,
c = 3
};
void f() {
  bit<8> good1;
  good1 = ((e.a) ++ (e.b));
}
