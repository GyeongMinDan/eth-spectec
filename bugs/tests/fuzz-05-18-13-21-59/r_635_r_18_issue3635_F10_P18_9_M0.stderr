fuzz/fuzz1/illtyped/r_18_issue3635_F10_P18_9_M0.p4(13): [--Werror=type-error] error: '(bool){}'
a = ((bool) ({ })),
     ^^^^^^^^^^^^
  ---- Actual error:
  fuzz/fuzz1/illtyped/r_18_issue3635_F10_P18_9_M0.p4(13): Cannot unify type 'tuple<>' with type 'bool'
  a = ((bool) ({ })),
               ^^^
  fuzz/fuzz1/illtyped/r_18_issue3635_F10_P18_9_M0.p4(13)
  a = ((bool) ({ })),
        ^^^^
  ---- Originating from:
  fuzz/fuzz1/illtyped/r_18_issue3635_F10_P18_9_M0.p4(13): Cannot cast from 'tuple<>' to 'bool'
  a = ((bool) ({ })),
               ^^^
  fuzz/fuzz1/illtyped/r_18_issue3635_F10_P18_9_M0.p4(13)
  a = ((bool) ({ })),
        ^^^^
In file: /p4c/frontends/p4/typeChecking/typeChecker.cpp:172
Compiler Bug: /p4c/frontends/p4/typeChecking/typeChecker.cpp:172: Null srcType

